actors/evm/src/interpreter/precompiles/bls12_381/g2_add.rs

```diff
@@ -0,0 +1,256 @@
+use crate::interpreter::{
+    System,
+    precompiles::{PrecompileContext, PrecompileError, PrecompileResult},
+};
+use fil_actors_runtime::runtime::Runtime;
+
+use crate::interpreter::precompiles::bls_util::{
+    G2_ADD_INPUT_LENGTH, G2_INPUT_ITEM_LENGTH, encode_g2_point, extract_g2_input, is_infinity,
+    p2_from_affine, p2_to_affine,
+};
+
+use blst::{blst_p2, blst_p2_add_or_double_affine, blst_p2_affine};
+
+/// **BLS12_G2ADD Precompile**
+///
+/// Implements G2 point addition according to [EIP-2537](https://eips.ethereum.org/EIPS/eip-2537).
+pub fn bls12_g2add<RT: Runtime>(
+    _: &mut System<RT>,
+    input: &[u8],
+    _: PrecompileContext,
+) -> PrecompileResult {
+    if input.len() != G2_ADD_INPUT_LENGTH {
+        return Err(PrecompileError::IncorrectInputSize);
+    }
+
+    // Split the input into two segments corresponding to the two G2 points.
+    let a_aff = extract_g2_input(&input[..G2_INPUT_ITEM_LENGTH], false)?;
+    let b_aff = extract_g2_input(&input[G2_INPUT_ITEM_LENGTH..], false)?;
```

Comment:
note: https://doc.rust-lang.org/std/primitive.slice.html#method.split_at is your friend here (and elsewhere).

Reply:
(although I haven't tried making the change, it might be worse)

Reply:
👍  fc764ef0